### PR TITLE
feat: improve navbar and topics accessibility

### PR DESCRIPTION
**Pull request by HussainTaj-arbisoft**
# What are the relevant tickets?
Closes https://github.com/mitodl/ocw-hugo-themes/issues/1220
Fixes https://github.com/mitodl/ocw-hugo-themes/issues/1237

# Description (What does it do?)

This PR improves the accessibility of the sidebars. In general, this PR:

* Converts relevant `<a>` tags to `<button>`s.
* Adds some extra semantic labels for the navbar.
* Makes the HTML more semantic by using `<ul>` for topics in the course info bar.
* Fixes the layout shift issue when topics were toggled.

# Screenshots (if appropriate):

https://github.com/mitodl/ocw-hugo-themes/assets/71316217/fa9893a5-6d74-4cd3-b644-56c4a8c49144

# How can this be tested?

1. Download content for the course [6.01sc-spring-2011](https://github.mit.edu/ocw-content-rc/6.01sc-spring-2011), and place it inside your local content directory. This step is optional if you have your setup configured to fetch remote content.
2. Navigate your local `ocw-hugo-themes` directory.
3. Checkout branch `hussaintaj/1220/expandables`.
4. Run
    ```
    yarn start course 6.01sc-spring-2011
    ```
5. Open https://localhost:3000/.
6. In the course info sidebar, on the right, toggle the collapsible topics/subtopics.
    **Expected Behavior**: No unexpected layout shifts occur.
7. Open your screen reading assistant (VoiceOver on Mac). Let it read the screen. (On Mac, you can use Ctrl + Option + A to start reading from the cursor position). Listen to the UI navbar on the left, and the topics section of the course info bar on the right. Expand items and listen to what the reader says when the items are expanded. Refer to the recording for more details.
    **Expected Behavior**: The voice assistant's speech is comprehensible.
    **Expected Behavior**: No decorative icon is spoken out loud.
    **Expected Behavior**: Accessibility works the same on mobile screen.
8. Repeat step 7 but this time build the course for offline use and run it from disk.
    ```
    yarn build /absolute/path/6.01sc-spring-2011 /absolute/path/ocw-hugo-projects/ocw-course-v2/config-offline.yaml
    ```

# Additional Context
Best read up on the source of it all https://github.com/mitodl/hq/issues/1918. In particular, this comment: https://github.com/mitodl/hq/issues/1918#issuecomment-1701002386

### CODE REVIEW

File: tests-e2e/ocw-ci-test-course/course-info.spec.ts [hunk 1]
```diff
@@ -14,3 +17,23 @@ test("Course info section can be toggled", async ({ page }) => {
...
+
+  await expect(heading).toBeVisible()
+  await expect(topicCollapseButton).toHaveAttribute("aria-expanded", "true")
+
+  await topicCollapseButton.click()
+  const mainSection = await page.locator(`#${MAIN_COURSE_SECTION_ID}`)
+  const drawer = await page.locator(`#${DESKTOP_COURSE_DRAWER_ID}`)
+
+  await expect(mainSection).toHaveClass(/.*col-lg-9.*/)
+  await expect(drawer).toHaveClass(/.*col-3.*/)
```
Comment (ChristopherChudzicki) [inline tests-e2e/ocw-ci-test-course/course-info.spec.ts:38]: This is for https://github.com/mitodl/ocw-hugo-themes/issues/1237#issuecomment-1700920199, right? 

Another option would be to compare `drawer.getBoundingClientRect().width` before / after the click, i guess.

**Review by ChristopherChudzicki — APPROVED** [review]
Works great, LGTM. I did leave some potential label wording suggestions—up to you.

File: course-v2/layouts/_default/baseof.html [hunk 2]
```diff
@@ -91,15 +91,20 @@
         document.addEventListener("DOMContentLoaded", () => {
           const drawer = $(`#${DESKTOP_COURSE_DRAWER_ID}`)
           const mainSection = $(`#${MAIN_COURSE_SECTION_ID}`)
-          
 
           drawer.on("shown.bs.collapse", (event) => {
+            if (event.target.id !== DESKTOP_COURSE_DRAWER_ID) {
+              return;
+            }
```
[the remark this answers is not in the document]
Comment (HussainTaj-arbisoft) [reply course-v2/layouts/_default/baseof.html:98]: This condition is activated when the topic collapsible is toggled.

We're catching `collapse` events on the drawer collapsible. The drawer collapsible has nested collapsibles (i.e. topics). When topics are collapsed/shown the event bubbles up and triggers our handler on the drawer collapsible.

This condition makes sure that layout changes only happen if the event's original target is the course-info collapsible.

File: tests-e2e/ocw-ci-test-course/course-info.spec.ts (hunk 1 above)
Comment (HussainTaj-arbisoft) [reply tests-e2e/ocw-ci-test-course/course-info.spec.ts:38]: Yes. I tried it. In that approach, we only need to check the x-coordinates before and after. Checking the classes felt more relevant to the bug.

**Pull request merged**